raiden/tests/integration/transfer/test_refundtransfer.py

```diff
@@ -182,3 +177,140 @@ def test_refund_transfer(raiden_chain, token_addresses, deposit, network_wait):
         app1, deposit - amount_path - amount_drain, [],
         app2, deposit + amount_path + amount_drain, [],
     )
+
+
+@pytest.mark.parametrize('privatekey_seed', ['test_refund_transfer:{}'])
+@pytest.mark.parametrize('number_of_nodes', [4])
+@pytest.mark.parametrize('number_of_tokens', [1])
+@pytest.mark.parametrize('channels_per_node', [CHAIN])
+def test_refund_transfer_after_2nd_hop(raiden_chain, token_addresses, deposit, network_wait):
+    """Test the refund transfer sent due to failure after 2nd hop"""
+    # Topology:
+    #
+    #  0 -> 1 -> 2 -> 3
+    #
+    app0, app1, app2, app3 = raiden_chain
+    token_address = token_addresses[0]
+    payment_network_identifier = app0.raiden.default_registry.address
+    token_network_identifier = views.get_token_network_identifier_by_token_address(
+        views.state_from_app(app0),
+        payment_network_identifier,
+        token_address,
+    )
+
+    # make a transfer to test the path app0 -> app1 -> app2 -> app3
+    identifier_path = 1
+    amount_path = 1
+    mediated_transfer(
```

Comment:
why do we have this extra transfer? it not necessary at all for the test

Reply:
Just to have a more realistic test where the path has already been used.